src/FixedSizeArrays.jl

```diff
@@ -38,6 +38,24 @@ end
 
 Base.isassigned(a::FixedSizeArray, i::Int) = isassigned(a.mem, i)
 
+# safe product of a tuple of integers, for calculating dimensions size
+
+checked_dims_impl(a::Int, ::Tuple{}) = a
+function checked_dims_impl(a::Int, t::Tuple{Int,Vararg{Int,N}}) where {N}
+    b = first(t)
+    (m, o) = Base.Checked.mul_with_overflow(a, b)
+    o && throw(ArgumentError("array dimensions too great, can't represent length"))
+    r = Base.tail(t)::NTuple{N,Int}
+    checked_dims_impl(m, r)::Int
+end
+
+checked_dims(::Tuple{}) = 1
```

Comment:
This method doesn't seem to be actually used?  Or are there constructors I'm missing that'd call it?

Reply:
This should call it, I'll add a test:

```julia-repl
julia> using FixedSizeArrays

julia> FixedSizeArray{Int,0}(undef, ())
0-dimensional FixedSizeArray{Int64, 0}:
0
```